packages/core/src/baseclient.ts

```diff
@@ -434,7 +497,15 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
           const isInternalException =
             hint && hint.data && (hint.data as { [key: string]: unknown }).__sentry__ === true;
           // We skip beforeSend in case of transactions
+
           if (isInternalException || !beforeSend || isTransaction) {
+            if (!isTransaction) {
```

Comment:
This is rather complicated to understand at a glance 😕 

Is there an obvious way to untangle the conditions here?

Reply:
Extracted this nasty `if` and left a comment to be more descriptive. I'm only not sure about L526-258, whether it shouldn't be moved earlier to L519.